go-controller/pkg/factory/factory.go

```diff
@@ -1295,6 +1299,12 @@ func (wf *WatchFactory) addHandler(objType reflect.Type, namespace string, sel l
 	intInf.Lock()
 	defer intInf.Unlock()
 
+	// we are going to add a handler, we need to update the atomic signal that handlers exist now
+	// so that we do not miss events after we list current items.
+	// We need to do this after we get internal informer lock, to preserve that we can be the only one updating
+	// the atomic and preserve known state of the atomic while the handler is going to be added in the future
```

Comment:
I'm not sure what you meant with "while the handler is going to be added in the future".  Did you mean that you want to store the old value so you can rollback in case of error?

Reply:
Yeah your text is clearer. I can fix the comment later in another pr